compiler/test/stdlib/path.test.gr

```diff
@@ -0,0 +1,286 @@
+import Path from "path"
+import Result from "result"
+import Option from "option"
+import List from "list"
+
+// let relFile = x =>
+//   Option.expect(
+//     "Expected " ++ x ++ " to be relative file",
+//     Path.relativeFile(x)
+//   )
+// let relDir = x =>
+//   Option.expect(
+//     "Expected " ++ x ++ " to be relative directory",
+//     Path.relativeDir(x)
+//   )
+// let absFile = x =>
+//   Option.expect(
+//     "Expected " ++ x ++ " to be absolute file",
+//     Path.absoluteFile(x)
+//   )
+// let absDir = x =>
+//   Option.expect(
+//     "Expected " ++ x ++ " to be absolute directory",
+//     Path.absoluteDir(x)
+//   )
+
+let fs = Path.fromString
+
+// test that a variety of properties hold for valid relative/absolute paths
+record ParseFileTestData {
+  pathStr: String,
+  expParent: String,
+  expStr: String,
+  expName: Option<String>,
+  expStem: String,
+  expExt: String,
+}
+
+let parseFileTests = [
+  {
+    pathStr: "./dir/file.txt",
+    expParent: "./dir/",
+    expStr: "./dir/file.txt",
+    expName: Some("file.txt"),
+    expStem: "file",
+    expExt: ".txt",
+  },
+  {
+    pathStr: "file",
+    expParent: ".",
+    expStr: "./file",
+    expName: Some("file"),
+    expStem: "file",
+    expExt: "",
+  },
+  {
+    pathStr: "a.tar.gz",
+    expParent: ".",
+    expStr: "./a.tar.gz",
+    expName: Some("a.tar.gz"),
+    expStem: "a.tar",
+    expExt: ".gz",
+  },
+  {
+    pathStr: "../../file",
+    expParent: "../..",
+    expStr: "../../file",
+    expName: Some("file"),
+    expStem: "file",
+    expExt: "",
+  },
+  {
+    pathStr: ".\\dir\\file.txt",
+    expParent: ".",
+    expStr: "./.\\dir\\file.txt",
```

Comment:
See discussion in discord about how we should add an escape hatch for people that need to parse Windows-style paths